debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -437,7 +438,7 @@ public void changeRecord(P partition,
 
     private final class BufferingSnapshotChangeRecordReceiver implements SnapshotReceiver<P> {
 
-        private Supplier<DataChangeEvent> bufferedEvent;
+        private AtomicReference<BufferedDataChangeEvent> bufferedEventRef = new AtomicReference<>(new BufferedDataChangeEvent());
```

Comment:
@jpechane 
to avoid a null check here:
queue.enqueue(bufferedEventRef.getAndSet(nextBufferedEvent).dataChangeEvent);

The queue does another null check in the enqueue method.